File: python/SSAFY 입과전/mid02.py
# 숫자 배열 회전

'''
[입력]
첫줄 - 테스트 케이스 개수 T
N
N*N행렬 
'''
def rotate90(N, board) :
    newBoard = [[0] * N for i in range(N)] 
    for i in range(N):
        for j in range(N): 
            newBoard[i][j] = str(board[N - 1 - j][i])
    return newBoard; 

T = int(input())

for test_case in range(1, T + 1):
    # 입력값 받기 
    N = int(input())
    board = []
    for i in range(N):
        board.append(list(map(int, input().split(" ")))); 
    
    # 배열 회전하기 
    degree90 = rotate90(N, board)
    degree180 = rotate90(N, degree90)
    degree270 = rotate90(N, degree180)

    # 출력
    print(f"#{test_case}")
    for i in range(N):
        print("".join(degree90[i]), end=" ")
        print("".join(degree180[i]), end=" ")
        print("".join(degree270[i]), end=" ")
        print()

    
    # 파이썬 리스트 초기화 
    '''
    1. board = [0] * N
    2. (리스트 컴프리핸션) board = [0 for i in range(N)]

    이중 배열 [[0] * 4] * 4  이 딴식으로 하지 말자 - 내부 배열의 참조값이 다 같음 
    '''

File: python/SSAFY 입과전/Mid02.java
import java.util.Arrays;
import java.util.Scanner;

public class Mid02 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
		int T;
		T=sc.nextInt();
		for(int test_case = 1; test_case <= T; test_case++) {
            // 입력하기 
		    int N = sc.nextInt(); 
            int[][] board = new int[N][N]; 
            for(int i = 0; i < N; i++) {
                for(int j = 0; j < N; j++) {
                    board[i][j] = sc.nextInt(); 
                }
            }

            // 회전시키기 
            int[][] degree90 = rotateBoard(N, board); 
            int[][] degree180 = rotateBoard(N, degree90); 
            int[][] degree270 = rotateBoard(N, degree180);
            
            // 출력하기 
            System.out.printf("#%d\n", test_case);
            for(int i = 0; i < N; i++) {
                String dg90 = Arrays.toString(degree90[i]).replaceAll("[\\[\\], ]", ""); 
                String dg180 = Arrays.toString(degree180[i]).replaceAll("[\\[\\], ]", ""); 
                String dg270 = Arrays.toString(degree270[i]).replaceAll("[\\[\\], ]", ""); 
                
                System.out.printf("%s %s %s", dg90, dg180, dg270);
                System.out.println();
            }

		}
    }

    public static int[][] rotateBoard(int n, int[][] board) {
        int[][] newBoard = new int[n][n]; 
        
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                newBoard[i][j] = board[n - 1 - j][i]; 
            }
        }
        return newBoard; 
    }
    
}
